editors/code/src/debug.ts

```diff
@@ -6,11 +6,20 @@ import type * as ra from "./lsp_ext";
 import { Cargo } from "./toolchain";
 import type { Ctx } from "./ctx";
 import { prepareEnv } from "./run";
-import { execute, isCargoRunnableArgs, unwrapUndefinable } from "./util";
+import { execute, isCargoRunnableArgs, log, unwrapUndefinable } from "./util";
 import type { Config } from "./config";
 
 const debugOutput = vscode.window.createOutputChannel("Debug");
 
+
+// Every debug session with the same id (restart-candidate) should be in this list
+let debugSessionIdBuffer: string[] = [];
+
+// Here we want to keep track on everything that's currently running
+const activeDebugSessionIds: string[] = [];
```

Comment:
This seems a bit odd, to my knowledge there can only ever be one active debug session in vscode

Reply:
While this does seem like a complex feature, it is supported to have multiple debug sessions :)

<img width="871" alt="image" src="https://github.com/user-attachments/assets/2a7b8d2a-052a-4b75-8582-6a4c91f1741e">